Panic when working with proxy

**Describe the bug**
I was working with proxy and doing different actions with it, then the following appeared in logs:
```
[2020-04-01T19:24:27+03:00] INFO readPacket [tp:020055]: Received packet. rt_id=6 size=1 type="ClosePacket"
[2020-04-01T19:24:27+03:00] DEBUG readPacket [tp:020055]: Awaiting packet...
[2020-04-01T19:24:27+03:00] INFO [router]: Handling packet with descriptor rPK:0263127ed85b348829e3917eace16c11f9d01243981f189fd26652414be4d3b6c7, lPK:020055fab8d7679d81aa8207968bfa9ad312ee7af21a319a75cd45ee0ffd2ea1ac, rPort:49155, lPort:3
[2020-04-01T19:24:27+03:00] INFO [router]: Received close packet for route ID 6
[2020-04-01T19:24:27+03:00] INFO [router]: Handling close packet with descriptor rPK:0263127ed85b348829e3917eace16c11f9d01243981f189fd26652414be4d3b6c7, lPK:020055fab8d7679d81aa8207968bfa9ad312ee7af21a319a75cd45ee0ffd2ea1ac, rPort:49155, lPort:3
[2020-04-01T19:24:27+03:00] INFO [router]: Got new remote close packet with size 1 and route ID 6. Using rule: APP(keyRtID:6, rPK:0263127ed85b348829e3917eace16c11f9d01243981f189fd26652414be4d3b6c7, lPK:020055fab8d7679d81aa8207968bfa9ad312ee7af21a319a75cd45ee0ffd2ea1ac, rPort:49155, lPort:3)
[2020-04-01T19:24:27+03:00] INFO [RouteGroup rPK:0263127ed85b348829e3917eace16c11f9d01243981f189fd26652414be4d3b6c7, lPK:020055fab8d7679d81aa8207968bfa9ad312ee7af21a319a75cd45ee0ffd2ea1ac, rPort:49155, lPort:3]: Got close packet with code 0
ERROR READING FROM APP CONN SERVER SIDE: EOF
[2020-04-01T19:24:27+03:00] INFO [router]: Got new remote packet with size 12 and route ID 6. Using rule: APP(keyRtID:6, rPK:0263127ed85b348829e3917eace16c11f9d01243981f189fd26652414be4d3b6c7, lPK:020055fab8d7679d81aa8207968bfa9ad312ee7af21a319a75cd45ee0ffd2ea1ac, rPort:49155, lPort:3)
[2020-04-01T19:24:27+03:00] INFO [RouteGroup rPK:0263127ed85b348829e3917eace16c11f9d01243981f189fd26652414be4d3b6c7, lPK:020055fab8d7679d81aa8207968bfa9ad312ee7af21a319a75cd45ee0ffd2ea1ac, rPort:49155, lPort:3]: Remote got closed, stopping keep-alive loop
[2020-04-01T19:24:27+03:00] ERROR [RouteGroup rPK:0263127ed85b348829e3917eace16c11f9d01243981f189fd26652414be4d3b6c7, lPK:020055fab8d7679d81aa8207968bfa9ad312ee7af21a319a75cd45ee0ffd2ea1ac, rPort:49155, lPort:3]: error updating activity of rule 5 error="rule of id 5 not found"
[2020-04-01T19:24:27+03:00] INFO [app_gateway:69946eb5-7965-4e21-a1b4-e3a5a7d2eaae]: Request processed. _elapsed="13.178µs" _method="CloseConn" _received="7:24PM" input=0xc002bf80f6
[2020-04-01T19:24:27+03:00] INFO [skysocks-client]: [2020-04-01T19:24:27.379343+03:00] ERROR [skysocks]: Copy error:read tcp [::1]:1080->[::1]:55991: use of closed network connection
panic: send on closed channel

goroutine 409 [running]:
github.com/SkycoinProject/skywire-mainnet/pkg/router.(*RouteGroup).handleDataPacket(0xc0000a91e0, 0xc002c06320, 0x13, 0x20, 0x3, 0x3)
	/Users/nkryuchkov/GolandProjects/skywire-mainnet/pkg/router/route_group.go:468 +0xfd
github.com/SkycoinProject/skywire-mainnet/pkg/router.(*RouteGroup).handlePacket(0xc0000a91e0, 0xc002c06320, 0x13, 0x20, 0x3, 0x3)
	/Users/nkryuchkov/GolandProjects/skywire-mainnet/pkg/router/route_group.go:461 +0xd3
github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).handleDataPacket(0xc000269680, 0x187e840, 0xc00002a040, 0xc002c06320, 0x13, 0x20, 0x100000000000000, 0x100000000000000)
	/Users/nkryuchkov/GolandProjects/skywire-mainnet/pkg/router/router.go:436 +0x526
github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).handleTransportPacket(0xc000269680, 0x187e840, 0xc00002a040, 0xc002c06320, 0x13, 0x20, 0xc002c08150, 0x16)
	/Users/nkryuchkov/GolandProjects/skywire-mainnet/pkg/router/router.go:390 +0x15f
github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).serveTransportManager.func1(0xc000269680, 0x187e840, 0xc00002a040, 0xc000446140, 0xc0005bc900, 0xc002c06320, 0x13, 0x20)
	/Users/nkryuchkov/GolandProjects/skywire-mainnet/pkg/router/router.go:316 +0x81
created by github.com/SkycoinProject/skywire-mainnet/pkg/router.(*router).serveTransportManager
	/Users/nkryuchkov/GolandProjects/skywire-mainnet/pkg/router/router.go:315 +0xd0
```

**Steps to Reproduce**
Not sure how to reproduce it, I was trying to make proxy work in browser to fix https://github.com/SkycoinProject/skywire-mainnet/issues/280. 

**Actual behavior**
Visor panics.

**Expected behavior**
There's no panic

**Additional context**
Commit `b57b9821c819f78b46cdfe7f9ebacb6ba000e0b9`

--- a/pkg/router/route_group.go
+++ b/pkg/router/route_group.go
@@ -367,18 +367,20 @@ func (rg *RouteGroup) keepAliveLoop(interval time.Duration) {
 				continue
 			}
 
-			rg.sendKeepAlive()
+			if err := rg.sendKeepAlive(); err != nil {
+				rg.logger.Warnf("Failed to send keepalive: %v", err)
+			}
 		}
 	}
 }
 
-func (rg *RouteGroup) sendKeepAlive() {
+func (rg *RouteGroup) sendKeepAlive() error {
 	rg.mu.Lock()
 	defer rg.mu.Unlock()
 
 	if len(rg.tps) == 0 || len(rg.fwd) == 0 {
 		// if no transports, no rules, then no keepalive
-		return
+		return nil
 	}
 
 	for i := 0; i < len(rg.tps); i++ {
@@ -390,14 +392,13 @@ func (rg *RouteGroup) sendKeepAlive() {
 		}
 
 		packet := routing.MakeKeepAlivePacket(rule.NextRouteID())
-		errCh := rg.writePacketAsync(context.Background(), tp, packet, rule.KeyRouteID())
 
-		go func() {
-			if err := <-errCh; err != nil {
-				rg.logger.WithError(err).Warnf("Failed to send keepalive")
-			}
-		}()
+		if err := rg.writePacket(context.Background(), tp, packet, rule.KeyRouteID()); err != nil {
+			return err
+		}
 	}
+
+	return nil
 }
 
 // Close closes a RouteGroup with the specified close `code`:
@@ -492,12 +493,9 @@ func (rg *RouteGroup) handleClosePacket(code routing.CloseCode) error {
 func (rg *RouteGroup) broadcastClosePackets(code routing.CloseCode) {
 	for i := 0; i < len(rg.tps); i++ {
 		packet := routing.MakeClosePacket(rg.fwd[i].NextRouteID(), code)
-		errCh := rg.writePacketAsync(context.Background(), rg.tps[i], packet, rg.fwd[i].KeyRouteID())
-		go func(tp *transport.ManagedTransport) {
-			if err := <-errCh; err != nil {
-				rg.logger.WithError(err).Errorf("Failed to send close packet to %s", tp.Remote())
-			}
-		}(rg.tps[i])
+		if err := rg.writePacket(context.Background(), rg.tps[i], packet, rg.fwd[i].KeyRouteID()); err != nil {
+			rg.logger.WithError(err).Errorf("Failed to send close packet to %s", rg.tps[i].Remote())
+		}
 	}
 }
 

--- a/pkg/router/router.go
+++ b/pkg/router/router.go
@@ -294,38 +294,25 @@ func (r *router) Serve(ctx context.Context) error {
 }
 
 func (r *router) serveTransportManager(ctx context.Context) {
-	var once sync.Once
-	done := make(chan struct{})
-
 	for {
-		select {
-		case <-done:
-			return
-		default:
-			packet, err := r.tm.ReadPacket()
-			if err != nil {
-				if err == transport.ErrNotServing {
-					r.logger.WithError(err).Info("Stopped reading packets")
-					return
-				}
-				r.logger.WithError(err).Error("Stopped reading packets due to unexpected error.")
+		packet, err := r.tm.ReadPacket()
+		if err != nil {
+			if err == transport.ErrNotServing {
+				r.logger.WithError(err).Info("Stopped reading packets")
 				return
 			}
 
-			go func(packet routing.Packet) {
-				if err := r.handleTransportPacket(ctx, packet); err != nil {
-					if err == transport.ErrNotServing {
-						once.Do(func() {
-							r.logger.WithError(err).Warnf("Stopped serving Transport.")
-							close(done)
-						})
+			r.logger.WithError(err).Error("Stopped reading packets due to unexpected error.")
+			return
+		}
 
-						return
-					}
+		if err := r.handleTransportPacket(ctx, packet); err != nil {
+			if err == transport.ErrNotServing {
+				r.logger.WithError(err).Warnf("Stopped serving Transport.")
+				return
+			}
 
-					r.logger.Warnf("Failed to handle transport frame: %v", err)
-				}
-			}(packet)
+			r.logger.Warnf("Failed to handle transport frame: %v", err)
 		}
 	}
 }

